pkg/utils/duration.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+)
+
+var re = regexp.MustCompile(`^([0-9]{1,5}(h|m|s|ms)){1,4}$`)
+
+func ParseDuration(s string) (*time.Duration, error) {
+	/*
+		parseDuration parses a GEP-2257 Duration format to a time object
+		Valid date units in time.Duration are "ns", "us" (or "µs"), "ms", "s", "m", "h"
+		Valid date units according to GEP-2257 are "h", "m", "s", "ms"
+
+		input: string
+
+		output: time.Duration
+
+		See https://gateway-api.sigs.k8s.io/geps/gep-2257/ for more details.
+	*/
+	if matched := re.MatchString(s); matched == false {
```

Comment:
```suggestion
	if !re.MatchString(s) {
```

Seems like there's no real point to the `matched` variable...